src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/DetectorOptions.java

```diff
@@ -228,9 +220,23 @@ public Builder suppressor(MatchSuppressor val) {
         public DetectorOptions build() {
             message = Optional.ofNullable(message).orElse("");
             suppressor = Optional.ofNullable(suppressor).orElse(NeverSuppress.INSTANCE);
+            computePattern();
```

Comment:
as for me, it looks better in a way like
`pattern = Optional.ofNullable(format).map(this::getPattern).orElse(null)`

Reply:
Much better indeed. I'm refactoring right now. Thanks.

Reply:
@strkkk , I had never used `Optional` class before, so I went to checkout it's api a little bit.
Seems like an elegant way to avoid null pointer exceptions. Thanks for making me discover this feature. I'm definitely using it from now on :)

Reply:
[pitest report](https://wilcoln.github.io/checkstyle-reports/pit-reports/detector-after/com.puppycrawl.tools.checkstyle.checks.regexp/DetectorOptions.java.html) has been updated, still good.